Multiple definition error across multiple include_cpp calls

## Expected Behavior
This shouldn't crash and the symbols should be properly mangled so they don't conflict across `include_cpp` calls.

## Actual Behavior

```
error: linking with `cc` failed: exit status: 1
  |
  = note: command omitted for brevity
  = note: /usr/bin/ld: /autocxx-err/target/debug/build/autocxx-err-1cd1cc8230fa4802/out/libmultifileerror.a(172f5ed9fec53081-gen2.o): in function `cxxbridge1$new_autocxx_autocxx_wrapper':
          /autocxx-err/target/debug/build/autocxx-err-1cd1cc8230fa4802/out/autocxx-build-dir/cxx/gen2.cxx:136: multiple definition of `cxxbridge1$new_autocxx_autocxx_wrapper'; /autocxx-err/target/debug/build/autocxx-err-1cd1cc8230fa4802/out/libmultifileerror.a(172f5ed9fec53081-gen0.o):/autocxx-err/target/debug/build/autocxx-err-1cd1cc8230fa4802/out/autocxx-build-dir/cxx/gen0.cxx:136: first defined here
          collect2: error: ld returned 1 exit status
```

It should be noted that the name of the files (e.x. `gen2.cxx`, `gen0.cxx`) is consistent after running `cargo clean` and then recompiling.

## Steps to Reproduce the Problem

`src/thing.h`:
```cpp
#pragma once

struct Thing {
    float id;

    Thing(float id) : id(id) {}
};
```

`src/item.h`:
```cpp
#pragma once

struct Item {
    float id;

    Item(float id) : id(id) {}
};
```

`src/main.rs`:
```rust
use autocxx::WithinUniquePtr;

autocxx::include_cpp! {
    #include "thing.h"
    name!(thing)
    safety!(unsafe)
    generate!("Thing")
}

autocxx::include_cpp! {
    #include "item.h"
    name!(item)
    safety!(unsafe)
    generate!("Item")
}

fn main() {
    let thing = thing::Thing::new(15.).within_unique_ptr();
    let item = item::Item::new(15.).within_unique_ptr();
}
```

`src/build.rs`:
```rust
use autocxx_build::Builder;
use miette::Result;

fn main() -> Result<()> {
    Builder::new("src/main.rs", ["src/"])
        .build()?
        .warnings(false)
        .compile("multifileerror");

    println!("cargo:rerun-if-changed=src/lib.rs");

    Ok(())
}
```

## Specifications

- Version: 0.23.1
- Platform: Ubuntu 22.10

--- a/engine/src/conversion/analysis/fun/mod.rs
+++ b/engine/src/conversion/analysis/fun/mod.rs
@@ -1261,7 +1261,10 @@ impl<'a> FnAnalyzer<'a> {
             } else {
                 "_"
             };
-            cxxbridge_name = make_ident(format!("{cxxbridge_name}{joiner}autocxx_wrapper"));
+            cxxbridge_name = make_ident(
+                self.config
+                    .uniquify_name_per_mod(&format!("{cxxbridge_name}{joiner}autocxx_wrapper")),
+            );
             let (payload, cpp_function_kind) = match fun.synthetic_cpp.as_ref().cloned() {
                 Some((payload, cpp_function_kind)) => (payload, cpp_function_kind),
                 None => match kind {

--- a/integration-tests/tests/integration_test.rs
+++ b/integration-tests/tests/integration_test.rs
@@ -11860,8 +11860,8 @@ fn test_issue_1229() {
         }
 
         fn main() {
-            let thing = thing::Thing::new(15.).within_unique_ptr();
-            let item = item::Item::new(15.).within_unique_ptr();
+            let _thing = thing::Thing::new(15.).within_unique_ptr();
+            let _item = item::Item::new(15.).within_unique_ptr();
         }
     };
 

--- a/integration-tests/tests/integration_test.rs
+++ b/integration-tests/tests/integration_test.rs
@@ -11827,6 +11827,47 @@ fn test_issue_1214() {
     run_test("", hdr, quote! {}, &["C"], &[]);
 }
 
+#[test]
+fn test_issue_1229() {
+    let hdr = indoc! {"
+    struct Thing {
+        float id;
+    
+        Thing(float id) : id(id) {}
+    };
+
+    struct Item {
+        float id;
+    
+        Item(float id) : id(id) {}
+    };
+    "};
+    let hexathorpe = Token![#](Span::call_site());
+    let rs = quote! {
+        use autocxx::WithinUniquePtr;
+
+        autocxx::include_cpp! {
+            #hexathorpe include "input.h"
+            name!(thing)
+            safety!(unsafe)
+            generate!("Thing")
+        }
+        autocxx::include_cpp! {
+            #hexathorpe include "input.h"
+            name!(item)
+            safety!(unsafe)
+            generate!("Item")
+        }
+
+        fn main() {
+            let thing = thing::Thing::new(15.).within_unique_ptr();
+            let item = item::Item::new(15.).within_unique_ptr();
+        }
+    };
+
+    do_run_test_manual("", hdr, rs, None, None).unwrap();
+}
+
 // Yet to test:
 // - Ifdef
 // - Out param pointers

